### PR TITLE
[BC-12] Continue BC-10 task. Separate for easier reviewing. 

### PR DESCRIPTION
**Pull request by CorrectACHU**
Added new: -Added GetTicket details controller. -added deleteTicket controller.

### CODE REVIEW

File: carInsuranceBackEnd/src/main/java/com/yakvel/carInsuranceBackEnd/controllers/user/UserTicketPagesController.java [hunk 1]
```diff
@@ -52,17 +54,48 @@ public ResponseEntity<Set<TicketDto>> getUserTickets() {
...
 
+    @DeleteMapping("/ticket/{ticketId}")
+    public ResponseEntity<String> deleteTicket(@PathVariable long ticketId) throws IOException {
+        Person person = (Person) SecurityContextHolder.getContext().getAuthentication().getPrincipal();
+
+        Ticket ticket = ticketRepository.findById(ticketId).orElse(null);
+        if (ticket == null) {
+            return ResponseEntity.badRequest().body("Ticket does not exist");
+        } else if (ticket.getTicketOwner().getId() != person.getId()) {
+            return ResponseEntity.badRequest().body("This ticket does not belong to current user");
```
Comment (rmaskalenka) [inline carInsuranceBackEnd/src/main/java/com/yakvel/carInsuranceBackEnd/controllers/user/UserTicketPagesController.java]: should be HTTP 403 Forbidden
From documentation on 403 error

> [403 Forbidden](https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/403)
The client does not have access rights to the content

File: carInsuranceBackEnd/src/main/java/com/yakvel/carInsuranceBackEnd/controllers/user/UserTicketPagesController.java [hunk 2]
```diff
@@ -52,17 +54,48 @@ public ResponseEntity<Set<TicketDto>> getUserTickets() {
...
 
-    @PostMapping("/ticket")
+    @GetMapping("/ticket/{ticketId}")
+    public ResponseEntity getTicketDetails(@PathVariable long ticketId) {
+        Person person = (Person) SecurityContextHolder.getContext().getAuthentication().getPrincipal();
+        Ticket ticket = ticketRepository.findById(ticketId).orElse(null);
+        if (ticket == null) {
+            return ResponseEntity.badRequest().body("Ticket does not exist");
+        } else if (ticket.getTicketOwner().getId() != person.getId()) {
+            return ResponseEntity.badRequest().body("This ticket does not belong to current user");
```
Comment (rmaskalenka) [inline carInsuranceBackEnd/src/main/java/com/yakvel/carInsuranceBackEnd/controllers/user/UserTicketPagesController.java]: Should be HTTP 403

**Review by rmaskalenka — CHANGES_REQUESTED** [review]

File: carInsuranceBackEnd/src/main/java/com/yakvel/carInsuranceBackEnd/controllers/user/UserTicketPagesController.java (hunk 1 above)
Comment (CorrectACHU) [reply carInsuranceBackEnd/src/main/java/com/yakvel/carInsuranceBackEnd/controllers/user/UserTicketPagesController.java]: done

File: carInsuranceBackEnd/src/main/java/com/yakvel/carInsuranceBackEnd/controllers/user/UserTicketPagesController.java (hunk 2 above)
Comment (CorrectACHU) [reply carInsuranceBackEnd/src/main/java/com/yakvel/carInsuranceBackEnd/controllers/user/UserTicketPagesController.java]: done

**Pull request merged**